user-console/src/main/java/org/pentaho/mantle/client/ui/BurgerMenuBar.java

```diff
@@ -0,0 +1,320 @@
+    // Menu Popup Panels with an additional class of pen-BurgerMenuBarPopup (beyond the gwt-MenuBarPopup)
+    setStylePrimaryName( "pen-BurgerMenuBar" );
+
+    // Recover base style names to allow sharing the CSS.
+    addStyleName( "gwt-MenuBar" );
+    addStyleName( "gwt-MenuBar-vertical" );
+  }
+
+  @Override
+  public void onBrowserEvent( Event event ) {
+    MenuItem selectedItemBefore = null;
+    int eventType = DOM.eventGetType( event );
+    switch ( eventType ) {
+      case Event.ONCLICK: {
+        onItemClick( findItemLocal( DOM.eventGetTarget( event ) ) );
+        return;
+      }
+
+      // Overriding mouse over and out, and prevent calling the base, private itemOver method,
+      // is the only way to prevent hovering-over from automatically open a sub-menu on non-top-level menus...
+      case Event.ONMOUSEOVER:
+        onItemOver( findItemLocal( DOM.eventGetTarget( event ) ), true );
+        return;
+
+      case Event.ONMOUSEOUT:
+        onItemOver( findItemLocal( DOM.eventGetTarget( event ) ), false );
+        return;
+
+      case Event.ONKEYDOWN:
+        switch ( event.getKeyCode() ) {
+          case KeyCodes.KEY_ESCAPE:
+            // Override so that only this menubar's popup menu is closed,
+            // and not all popups until (but except) the top-level one.
+            onEscape();
+            eatEventLocal( event );
+            return;
+
+          case KeyCodes.KEY_ENTER:
+            // Override to also place focus on submenu, when there is one.
+            onEnter();
+            eatEventLocal( event );
+            return;
+
+          default:
+            selectedItemBefore = getSelectedItem();
+            break;
+        }
+        break;
+
+      default:
+        break;
+    }
+
+    super.onBrowserEvent( event );
+
+    if ( eventType == Event.ONKEYDOWN ) {
+      MenuItem selectedItemAfter = getSelectedItem();
+      if ( selectedItemAfter != null && selectedItemAfter != selectedItemBefore ) {
+        // Selected menu item changed when using the keyboard.
+        ensureVisible( selectedItemAfter );
+      }
+    }
+  }
+
+  private void onItemClick( MenuItem item ) {
+    focus();
+
+    if ( item != null ) {
+      doItemActionLocal( item, true, true );
+    }
+  }
+
+  private void onItemOver( MenuItem item, boolean isOver ) {
+    // Still need to select and unselect items.
+    if ( item != null && item.isEnabled() ) {
+      selectItem( isOver ? item : null );
+      focus();
+    }
+  }
+
+  protected void onEscape() {
+    closeLocal( true );
+  }
+
+  protected void onEnter() {
+    if ( !selectFirstItemIfNoneSelectedLocal() ) {
+      doItemActionLocal( getSelectedItem(), true, true );
+    }
+  }
+
+  protected void ensureVisible( MenuItem menuItem ) {
+    ElementUtils.scrollVerticallyIntoView( menuItem.getElement() );
+  }
+
+  public List<UIObject> getAllItems() {
+    return MenuBarUtils.getAllItems( this );
+  }
+
+  @Override
+  public List<MenuItem> getItems() {
```

Comment:
Should this be removed BurgerMenuBar#getItems()?
what's the point of overriding the super class implementation and then only calling the super class implementation?